lib/index.js

```diff
@@ -0,0 +1,127 @@
+'use strict';
+
+const path = require('path');
+const Filter = require('broccoli-persistent-filter');
+const md5Hex = require('md5-hex');
+const extensions = require('markdown-extensions');
+const stringify = require('json-stable-stringify');
+const testGenerators = require('./test-generators');
+const remarkEngine = require('./remark-engine');
+const testGeneratorNames = Object.keys(testGenerators);
+
+function resolveInputDirectory(inputNode) {
+	if (typeof inputNode === 'string') {
+		if (path.isAbsolute(inputNode)) {
+			return inputNode;
+		}
+		return path.join(process.cwd(), inputNode);
+	}
+
+	// eslint-disable-next-line no-underscore-dangle
+	const nodeInfo = inputNode.__broccoliGetInfo__();
+
+	if (nodeInfo.nodeType === 'source') {
+		return nodeInfo.sourceDirectory;
+	}
+
+	if (nodeInfo.inputNodes.length > 1) {
+		// eslint-disable-next-line max-len
+		throw new Error('BroccoliRemark can only handle one:* broccoli nodes, but part of the given input pipeline is a many:* node.' +
+			'(broccoli-merge-trees is an example of a many:* node) Please perform many:* operations after linting.');
+	}
+
+	return resolveInputDirectory(nodeInfo.inputNodes[0]);
+}
+
+function BroccoliRemark(inputNode, _options) {
+	if (!(this instanceof BroccoliRemark)) {
+		return new BroccoliRemark(inputNode, _options);
+	}
+
+	const options = _options || {};
+
+	if (!options.hasOwnProperty('persist')) {
```

Comment:
And merge default options with customOptions. I think Its more clear. Something like this.
```
const defaultOptions = {
    quiet: true,
    persist: true
};
const userOptions = {quiet: false};

const options = {...defaultOptions, ...userOptions};
```